aea/protocols/generator.py

```diff
@@ -919,216 +970,958 @@ def _message_class_str(self) -> str:
+        )
+        cls_str += (
+            indent
+            + "- Dialogue: The dialogue class maintains state of a dialogue and manages it.\n"
+        )
+        cls_str += (
+            indent + "- Dialogues: The dialogues class keeps track of all dialogues.\n"
+        )
+        cls_str += indent + '"""\n\n'
+
+        # Imports
+        cls_str += indent + "from enum import Enum\n"
+        cls_str += indent + "from typing import Dict, List, Tuple, Union, cast\n\n"
+        cls_str += (
+            indent
+            + "from aea.helpers.dialogue.base import Dialogue, DialogueLabel, Dialogues\n"
+        )
+        cls_str += indent + "from aea.mail.base import Address\n"
+        cls_str += indent + "from aea.protocols.base import Message\n\n"
+        cls_str += indent + "from {}.message import {}Message\n".format(
+            self.path_to_protocol_package, self.protocol_specification_in_camel_case,
+        )
+
+        # Constants
+        cls_str += indent + "\n"
+        cls_str += indent + "REPLY = " + str(self._reply) + "\n"
+
+        # Class Header
+        cls_str += "\n\nclass {}Dialogue(Dialogue):\n".format(
+            self.protocol_specification_in_camel_case
+        )
+        _change_indent(1)
+        cls_str += (
+            indent
+            + '"""The {} dialogue class maintains state of a dialogue and manages it."""\n'.format(
+                self.protocol_specification.name
+            )
+        )
+
+        # Class Constants
+        cls_str += indent + "\n"
+        cls_str += indent + "STARTING_MESSAGE_ID = 1\n"
+        cls_str += indent + "STARTING_TARGET = 0\n"
+        cls_str += indent + "\n"
+
+        # Performatives Enum
+        cls_str += "\n" + self._end_state_enum_str()
+        cls_str += "\n" + self._agent_role_enum_str()
+        cls_str += "\n"
+
+        # __init__
+        cls_str += indent + "def __init__(\n"
+        _change_indent(1)
+        cls_str += (
+            indent + "self, dialogue_label: DialogueLabel, is_seller: bool, **kwargs\n"
+        )
+        _change_indent(-1)
+        cls_str += indent + ") -> None:\n"
+        _change_indent(1)
+        cls_str += indent + '"""\n'
+        cls_str += indent + "Initialize a dialogue label.\n\n"
+        cls_str += indent + ":param dialogue_label: the identifier of the dialogue.\n"
+        cls_str += (
+            indent
+            + ":param is_seller: indicates whether the agent associated with the dialogue is a seller or buyer\n\n"
+        )
+        cls_str += indent + ":return: None\n"
+        cls_str += indent + '"""\n'
+        cls_str += indent + "super().__init__(self, dialogue_label=dialogue_label)\n"
+        cls_str += indent + "self._is_seller = is_seller\n"
+        cls_str += indent + "self._role = (\n"
+        _change_indent(1)
+        cls_str += (
+            indent
+            + "{}Dialogue.AgentRole.SELLER if is_seller else {}Dialogue.AgentRole.BUYER\n".format(
```

Comment:
Roles should be configurable, see above.

Reply:
Done